fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppState.kt

```diff
@@ -64,6 +65,7 @@ data class AppState(
     val collections: List<TabCollection> = emptyList(),
     val expandedCollections: Set<Long> = emptySet(),
     val mode: BrowsingMode = BrowsingMode.Normal,
+    val selectedTabId: String? = null,
```

Comment:
Should we really have this in the `AppState` if it's already known in the `BrowserStore`?

Reply:
Good question! This is required for the line in the binding [here](https://github.com/mozilla-mobile/firefox-android/blob/cfb8dc0f934a06be117e8d5d15c2e599275c2eac/fenix/app/src/main/java/org/mozilla/fenix/components/appstate/bindings/BrowserStoreBinding.kt#L31). Since `appState.mode` and `browserState.selectedTabId` are are not necessarily in sync at all times. For example, if you navigate to a tab, go home, and click the PBM button the selected tab will still be set to the tab, but the mode will be updated. 

This means that we need a way to track actual changes to the selected tab, so we can prevent lifecycle events from recreating the binding, triggering the initial observation, and overwriting the `appState.mode` with the mode from the selected tab. 

I'm open to other suggestions on how to avoid that, but this was what I came up.

I am saving this as an example for why we should move to a single store model eventually, as we could naturally avoid this problem under that paradigm.

Reply:
Just looking at the [comment](https://github.com/mozilla-mobile/firefox-android/blob/cfb8dc0f934a06be117e8d5d15c2e599275c2eac/fenix/app/src/main/java/org/mozilla/fenix/components/appstate/bindings/BrowserStoreBinding.kt#L29-30) about re-emission, I am wondering what did you observe from re-emission.

Reply:
if `AppState.selectedTabId` is safe to consume, some of the `TabsTray` Composable APIs could be simplified as a result

Reply:
@MozillaNoah Would be curious for more details there, but I would hesitate to suggest rushing for replacements just yet. Long-term, we may want to combine these Stores instead of replicate this state between them

Reply:
I agree let's not rush this. 

> Long-term, we may want to combine these Stores instead of replicate this state between them
I think I would like to see more details here because I simply don't see how we want consumer application state inside of `BrowserStore`, which could be consumed by any external app.

Reply:
TLDR: `BrowserStore` is being passed around as a dependency for the tab lists _solely_ for obtaining the `selectedTabId`, making the root `TabsTray` composable dependent on three stores 😕

Reply:
How come we don't just pass down only the `selectedTabId`?